src/lib/walletLinking/chainUtils.ts

```diff
@@ -1,10 +1,13 @@
 import EthereumIcon from "@/components/icons/EthereumIcon";
 import SolanaIcon from "@/components/icons/SolanaIcon";
-import { isAddress } from "viem";
+
+// Module-level caches for imported validators to optimize repeated calls
+let viemValidator: ((address: string) => boolean) | null = null;
+let solanaValidator: ((address: string) => boolean) | null = null;
```

Comment:
Why do we need this? AFAIK the dynamic imports are cached automatically by the es module system. Was there some sort of measurable performance issue this resolved?